leshan-server-core/src/main/java/org/eclipse/leshan/server/impl/LwM2mRequestSenderImpl.java

```diff
@@ -101,65 +75,20 @@ public LwM2mRequestSenderImpl(LwM2mRequestSender defaultRequestSender, QueuedReq
-        defaultRequestSender.removeResponseListener(this.defaultResponseListener);
-        processingExecutor.shutdown();
-    }
-
-    private ResponseListener createBindingAwareResponseListener() {
-        return new ResponseListener() {
-
-            @Override
-            public void onResponse(String clientEndpoint, String requestTicket, LwM2mResponse response) {
-                LOG.debug("Response received for request: {}", requestTicket);
-                Client registrationInfo = clientRegistry.get(clientEndpoint);
-                // process only when client is still known
-                if (registrationInfo != null) {
-                    // only if the client has used Queue mode notify its
-                    // listener
-                    if (registrationInfo.usesQueueMode()) {
-                        LOG.debug("Notifying QueueModeResponseListener.onResponse for ticket: {}", requestTicket);
-                        queueModeResponseListener.onResponse(clientEndpoint, requestTicket, response);
-                    }
-                    LOG.debug("Notifying All ResponseListeners.onResponse for ticket: {}", requestTicket);
-                    processingExecutor.execute(
-                            new ResponseProcessingTask(clientEndpoint, requestTicket, responseListeners, response));
-                }
-            }
-
-            @Override
-            public void onError(String clientEndpoint, String requestTicket, Exception exception) {
-                LOG.info("Exception on sending the request: {}", requestTicket, exception);
-                Client registrationInfo = clientRegistry.get(clientEndpoint);
-                // process only when client is still known
-                if (registrationInfo != null) {
-                    // only if the client uses Queue mode notify its listener
-                    if (registrationInfo.usesQueueMode()) {
-                        LOG.debug("Notifying QueueModeResponseListener.onError for ticket: {}", requestTicket);
-                        queueModeResponseListener.onError(clientEndpoint, requestTicket, exception);
-                    }
-
-                    // When client uses no queue mode and no timeout exception,
-                    // notify all listeners.
-                    // Otherwise queue mode will do retry sometime later when
-                    // client connects again.
-                    if (!(exception instanceof TimeoutException) && !(registrationInfo.usesQueueMode())) {
-                        LOG.debug("Notifying All ResponseListeners.onError for ticket: {}", requestTicket);
-                        processingExecutor.execute(new ResponseProcessingTask(clientEndpoint, requestTicket,
-                                responseListeners, exception));
-                    }
-                }
-            }
-        };
-
+        if (queuedRequestSender instanceof Stoppable) {
+            ((Stoppable) queuedRequestSender).stop();
+        }
```

Comment:
Thought of it and left it as it is not an instance of Stoppable. No problem. I can add it.